app/routes/next/project/$slug/settings/web-social.tsx

```diff
@@ -29,18 +62,260 @@ export const loader = async (args: DataFunctionArgs) => {
     return redirect(redirectPath, { headers: response.headers });
   }
 
-  return null;
+  const project = await prismaClient.project.findUnique({
+    select: {
+      website: true,
+      facebook: true,
+      linkedin: true,
+      xing: true,
+      twitter: true,
+      mastodon: true,
+      blueSky: true,
+      instagram: true,
+      youtube: true,
+    },
+    where: {
+      slug: params.slug,
+    },
+  });
+  invariantResponse(project !== null, "Project not found", {
+    status: 404,
+  });
+
+  return json({ project });
 };
 
+export async function action({ request, params }: DataFunctionArgs) {
+  const response = new Response();
+  const authClient = createAuthClient(request, response);
+  const sessionUser = await getSessionUser(authClient);
+  // check slug exists (throw bad request if not)
+  invariantResponse(params.slug !== undefined, "No valid route", {
+    status: 400,
+  });
+  const redirectPath = await getRedirectPathOnProtectedProjectRoute({
+    request,
+    slug: params.slug,
+    sessionUser,
+    authClient,
+  });
+  if (redirectPath !== null) {
+    return redirect(redirectPath, { headers: response.headers });
+  }
+  // Validation
+  const formData = await request.formData();
+  const submission = await parse(formData, {
+    schema: (intent) =>
+      webSocialSchema.transform(async (data, ctx) => {
+        if (intent !== "submit") return { ...data };
```

Comment:
please wrap in curly braces and use new line
reason: extending this statement will create a change on git on the if statement which wasn't actually changed

```
// intial
if (...) return ...;

// changes
if (...) {
  the change;
  return ...;
}
```

better:
```
// initial
if (...) {
  return ...;
}

// changes
...
  the change;
...

```